src/build.sh

```diff
@@ -99,12 +110,17 @@ function kernel_build()
   fi
 
   if [[ -n "${options_values['CCACHE']}" ]]; then
-    optimizations="CC=\"ccache gcc -fdiagnostics-color\" $parallel_cores"
+    if [[ -n "$llvm" ]]; then
+      ccache='CC="ccache clang -fdiagnostics-color"'
+    else
+      ccache='CC="ccache gcc -fdiagnostics-color"'
+    fi
```

Comment:
Maybe to avoid coding duplication, we could have a' compiler' variable that uses gcc by default (and you can set `clang` inside the `$llvm` condition). After that, you just need to replace the default compiler and build the `ccache` based on that (e.g., `ccache $compiler -fdiagnostics-color"`).

Reply:
really nice suggestion! thanks :)